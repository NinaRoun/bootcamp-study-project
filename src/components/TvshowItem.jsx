import React from 'react';
import style from '../stylesheets/main.styl';
import PropTypes from "prop-types";

const TvshowItem = ({tvshow, search}) => {

    const imgSrc = 'https://image.tmdb.org/t/p/w200' + tvshow.poster_path;
    const searchEscaped = search.replace(/[\\[.+*?(){|^$]/g, "\\$&");

    const getResultWithSearch = (searchedArea) => {

        const arr = searchedArea && searchedArea.split(new RegExp(searchEscaped, "i"));
        let symbolsParsed = 0;
        let result = [];
        for(let i = 0; i < arr.length; i++) {
            result.push(arr[i]);
            result.push((<span key={i} id="searchedArea" className={style.searchHighlighting}>{searchedArea.substr(arr[i].length + symbolsParsed, search.length)}</span>));
            symbolsParsed += (arr[i].length + search.length);
        }
        return result;

    };

    return (
        <tr>

            <th id="tv-id"> â„– {tvshow.id} </th>

            <th>
                <h2 id="tv-name"> {getResultWithSearch(tvshow.name)} </h2>
                <h3 id="tv-origin"> Country: { tvshow.origin_country } </h3>
                <h3 id="tv-language"> Original Language: { tvshow.original_language } </h3>
                <p id="tv-overview"> { getResultWithSearch(tvshow.overview) } </p>
            </th>

            <th> <img src={imgSrc} alt={tvshow.name}/> </th>

            <th>
                <span id="tv-vote"> Rate: { tvshow.vote_average }/10 </span>
                <p id="tv-vote-count"> Voted { tvshow.vote_count } times </p>
             </th>
        </tr>
    );
}

TvshowItem.propTypes = {
    tvshow: PropTypes.object.isRequired
}

export default TvshowItem;

