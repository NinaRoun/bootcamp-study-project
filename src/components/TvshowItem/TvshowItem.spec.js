import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TvshowItem from './TvshowItem.jsx';

configure({ adapter: new Adapter() });

function setup() {
    const props = {
        tvshow: {
            id: 1,
            name: 'one',
            origin_country: 'US',
            original_language: 'English',
            overview: 'Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.',
            vote_average: 7.3,
            vote_count: 250,
            first_air_date: "2020-03-21"
        },
        search: "searchedArea"
    };

    const enzymeWrapper = shallow(<TvshowItem {...props} />);

    return {
        props,
        enzymeWrapper
    }
}

describe('components', () => {
    describe('TvshowItem', () => {
        it('should render self and subcomponents', () => {
            const { enzymeWrapper } = setup();
            expect(enzymeWrapper).toMatchSnapshot();
            expect(enzymeWrapper.find('img')).toHaveLength(1);
        });

        it('simulates search', () => {
            const { enzymeWrapper, props } = setup();
            const searchComponent = enzymeWrapper.find('#searchedArea').at(1);
            expect(searchComponent.text()).toBe(props.search);

        })
    })
});


