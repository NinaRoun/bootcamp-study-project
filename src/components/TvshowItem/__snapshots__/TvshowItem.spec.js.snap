// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TvshowItem should render self and subcomponents 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TvshowItem
    search="searchedArea"
    tvshow={
      Object {
        "first_air_date": "2020-03-21",
        "id": 1,
        "name": "one",
        "origin_country": "US",
        "original_language": "English",
        "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
        "vote_average": 7.3,
        "vote_count": 250,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <th
          id="tv-id"
        >
           № 
          1
           
        </th>,
        <th>
          <h2
            id="tv-name"
          >
             
            one
            <span
              id="searchedArea"
            >
              
            </span>
             
          </h2>
          <h3
            id="tv-origin"
          >
             Country: 
            US
             
          </h3>
          <h3
            id="tv-language"
          >
             Original Language: 
            English
             
          </h3>
          <p
            id="tv-overview"
          >
             
            Lorem ipsum dolor sit amet, 
            <span
              id="searchedArea"
            >
              searchedArea
            </span>
             consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
            <span
              id="searchedArea"
            >
              
            </span>
             
          </p>
        </th>,
        <th>
           
          <img
            alt="one"
            src="https://image.tmdb.org/t/p/w200undefined"
          />
           
        </th>,
        <th>
          <span
            id="tv-vote"
          >
             Rate: 
            7.3
            /10 
          </span>
          <p
            id="tv-vote-count"
          >
             Voted 
            250
             times 
          </p>
        </th>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " № ",
            1,
            " ",
          ],
          "id": "tv-id",
        },
        "ref": null,
        "rendered": Array [
          " № ",
          1,
          " ",
        ],
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              id="tv-name"
            >
               
              one
              <span
                id="searchedArea"
              >
                
              </span>
               
            </h2>,
            <h3
              id="tv-origin"
            >
               Country: 
              US
               
            </h3>,
            <h3
              id="tv-language"
            >
               Original Language: 
              English
               
            </h3>,
            <p
              id="tv-overview"
            >
               
              Lorem ipsum dolor sit amet, 
              <span
                id="searchedArea"
              >
                searchedArea
              </span>
               consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
              <span
                id="searchedArea"
              >
                
              </span>
               
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                Array [
                  "one",
                  <span
                    id="searchedArea"
                  >
                    
                  </span>,
                ],
                " ",
              ],
              "id": "tv-name",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "one",
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": undefined,
                  "id": "searchedArea",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
              " ",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Country: ",
                "US",
                " ",
              ],
              "id": "tv-origin",
            },
            "ref": null,
            "rendered": Array [
              " Country: ",
              "US",
              " ",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Original Language: ",
                "English",
                " ",
              ],
              "id": "tv-language",
            },
            "ref": null,
            "rendered": Array [
              " Original Language: ",
              "English",
              " ",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                Array [
                  "Lorem ipsum dolor sit amet, ",
                  <span
                    id="searchedArea"
                  >
                    searchedArea
                  </span>,
                  " consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                  <span
                    id="searchedArea"
                  >
                    
                  </span>,
                ],
                " ",
              ],
              "id": "tv-overview",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Lorem ipsum dolor sit amet, ",
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "searchedArea",
                  "className": undefined,
                  "id": "searchedArea",
                },
                "ref": null,
                "rendered": "searchedArea",
                "type": "span",
              },
              " consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": undefined,
                  "id": "searchedArea",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
              " ",
            ],
            "type": "p",
          },
        ],
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <img
              alt="one"
              src="https://image.tmdb.org/t/p/w200undefined"
            />,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "one",
              "src": "https://image.tmdb.org/t/p/w200undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          " ",
        ],
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              id="tv-vote"
            >
               Rate: 
              7.3
              /10 
            </span>,
            <p
              id="tv-vote-count"
            >
               Voted 
              250
               times 
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Rate: ",
                7.3,
                "/10 ",
              ],
              "id": "tv-vote",
            },
            "ref": null,
            "rendered": Array [
              " Rate: ",
              7.3,
              "/10 ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Voted ",
                250,
                " times ",
              ],
              "id": "tv-vote-count",
            },
            "ref": null,
            "rendered": Array [
              " Voted ",
              250,
              " times ",
            ],
            "type": "p",
          },
        ],
        "type": "th",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <th
            id="tv-id"
          >
             № 
            1
             
          </th>,
          <th>
            <h2
              id="tv-name"
            >
               
              one
              <span
                id="searchedArea"
              >
                
              </span>
               
            </h2>
            <h3
              id="tv-origin"
            >
               Country: 
              US
               
            </h3>
            <h3
              id="tv-language"
            >
               Original Language: 
              English
               
            </h3>
            <p
              id="tv-overview"
            >
               
              Lorem ipsum dolor sit amet, 
              <span
                id="searchedArea"
              >
                searchedArea
              </span>
               consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
              <span
                id="searchedArea"
              >
                
              </span>
               
            </p>
          </th>,
          <th>
             
            <img
              alt="one"
              src="https://image.tmdb.org/t/p/w200undefined"
            />
             
          </th>,
          <th>
            <span
              id="tv-vote"
            >
               Rate: 
              7.3
              /10 
            </span>
            <p
              id="tv-vote-count"
            >
               Voted 
              250
               times 
            </p>
          </th>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " № ",
              1,
              " ",
            ],
            "id": "tv-id",
          },
          "ref": null,
          "rendered": Array [
            " № ",
            1,
            " ",
          ],
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                id="tv-name"
              >
                 
                one
                <span
                  id="searchedArea"
                >
                  
                </span>
                 
              </h2>,
              <h3
                id="tv-origin"
              >
                 Country: 
                US
                 
              </h3>,
              <h3
                id="tv-language"
              >
                 Original Language: 
                English
                 
              </h3>,
              <p
                id="tv-overview"
              >
                 
                Lorem ipsum dolor sit amet, 
                <span
                  id="searchedArea"
                >
                  searchedArea
                </span>
                 consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
                <span
                  id="searchedArea"
                >
                  
                </span>
                 
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  Array [
                    "one",
                    <span
                      id="searchedArea"
                    >
                      
                    </span>,
                  ],
                  " ",
                ],
                "id": "tv-name",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "one",
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": undefined,
                    "id": "searchedArea",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "span",
                },
                " ",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Country: ",
                  "US",
                  " ",
                ],
                "id": "tv-origin",
              },
              "ref": null,
              "rendered": Array [
                " Country: ",
                "US",
                " ",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Original Language: ",
                  "English",
                  " ",
                ],
                "id": "tv-language",
              },
              "ref": null,
              "rendered": Array [
                " Original Language: ",
                "English",
                " ",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  Array [
                    "Lorem ipsum dolor sit amet, ",
                    <span
                      id="searchedArea"
                    >
                      searchedArea
                    </span>,
                    " consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    <span
                      id="searchedArea"
                    >
                      
                    </span>,
                  ],
                  " ",
                ],
                "id": "tv-overview",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Lorem ipsum dolor sit amet, ",
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "searchedArea",
                    "className": undefined,
                    "id": "searchedArea",
                  },
                  "ref": null,
                  "rendered": "searchedArea",
                  "type": "span",
                },
                " consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": undefined,
                    "id": "searchedArea",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "span",
                },
                " ",
              ],
              "type": "p",
            },
          ],
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <img
                alt="one"
                src="https://image.tmdb.org/t/p/w200undefined"
              />,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "one",
                "src": "https://image.tmdb.org/t/p/w200undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            " ",
          ],
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                id="tv-vote"
              >
                 Rate: 
                7.3
                /10 
              </span>,
              <p
                id="tv-vote-count"
              >
                 Voted 
                250
                 times 
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Rate: ",
                  7.3,
                  "/10 ",
                ],
                "id": "tv-vote",
              },
              "ref": null,
              "rendered": Array [
                " Rate: ",
                7.3,
                "/10 ",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Voted ",
                  250,
                  " times ",
                ],
                "id": "tv-vote-count",
              },
              "ref": null,
              "rendered": Array [
                " Voted ",
                250,
                " times ",
              ],
              "type": "p",
            },
          ],
          "type": "th",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
